bouncer/sentry.py

```
@@ -0,0 +1,31 @@
+"""Report exceptions to Sentry/Raven."""
+import raven
+
+
+def raven_tween_factory(handler, _):
+    """Return a tween that reports uncaught exceptions to Sentry."""
+    def raven_tween(request):
+        """Report uncaught exceptions to Sentry."""
+        try:
+            return handler(request)
+        except:
+            request.raven.captureException()
+            raise
+
+    return raven_tween
+
+
+def raven_client(request):
+    """Return the raven.Client object."""
+    request.add_finished_callback(
+        lambda request: request.raven.context.clear(),
+    )
+
+    return raven.Client(
+        dsn=request.registry.settings.get("sentry_dsn"),
+    )
+
+
+def includeme(config):
+    config.add_request_method(raven_client, name="raven", reify=True)
+    config.add_tween("bouncer.sentry.raven_tween_factory")
```

Ah, although I note that we actually don't have any catch-all exception views in this application. So actually this is probably fine for now.